pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -668,6 +668,15 @@ SocketAddress serverAddrees() {
                 future.completeExceptionally(writeFuture.cause());
             }
         });
+        eventLoopGroup.schedule(() -> {
```

Comment:
I have a performance related concern: 

1) this would be scheduled per request. so it would end up with a lot of timeout tasks, no?
2) there is no mechanism to cancel the timeout tasks. so all tasks will be remaining until they are timed out, no?

Can we change the approach to have one timer task that peridocally check the pending requests list?

Reply:
> this would be scheduled per request. so it would end up with a lot of timeout tasks, 

in ideal scenario, application will have max few hundred producers/consumers and in worst case, 1 timeout task per producer/consumer so, it should not have large number of pending tasks. So, not sure if that will be create any performance impact.

> there is no mechanism to cancel the timeout tasks. so all tasks will be remaining until they are timed out. Can we change the approach to have one timer task that peridocally check the pending requests list?

umm.. that makes sense. Let me fix it.